internal/httphandler/utils.go

```diff
@@ -36,6 +36,29 @@ func extractVerifiedQueryPayload(w http.ResponseWriter, r *http.Request, queryTy
 			DbName: params["dbname"],
 			Key:    params["key"],
 		}
+	case constants.GetDataRange:
+		limit, err := strconv.Atoi(params["limit"])
+		if err != nil {
+			utils.SendHTTPResponse(w, http.StatusBadRequest, &types.HttpResponseErr{ErrMsg: err.Error()})
+			return nil, true
+		}
+
+		if limit < 0 {
+			utils.SendHTTPResponse(
+				w,
+				http.StatusBadRequest,
+				&types.HttpResponseErr{ErrMsg: "limit value " + strconv.Itoa(limit) + " should be greater than or equal to zero"},
+			)
+			return nil, true
+		}
+
+		payload = &types.GetDataRangeQuery{
+			UserId:   querierUserID,
+			DbName:   params["dbname"],
+			StartKey: params["startkey"][1 : len(params["startkey"])-1],
+			EndKey:   params["endkey"][1 : len(params["endkey"])-1],
```

Comment:
I am not sure what these two lines do... why chop the first char in the key?

Reply:
the start and end key string would be double quoted as ?`startkey=""&endkey="key2"&limit=10`. Here, we are removing the quotes. Here, I follow the CouchDB range query syntax.